lib/tasks/work_index_jobs.rake

```diff
@@ -2,38 +2,39 @@ desc <<-DESC.strip_heredoc
   Pulls an index job from an SQS queue and works it.
 DESC
 task work_index_jobs: :environment do
-  Rails.logger.info { "Starting work_index_jobs..." }
+  Rails.logger.info { "work_index_jobs: Started" }
 
   instance = OpenStax::Aws::AutoScalingInstance.me
   work_index_job = WorkIndexJobs.new
 
+  # If exceptions happen in the loop below, we need to rescue them and keep looping,
+  # because if we don't, our instance will never be able to get out of an eventual
+  # Terminating:Wait state.
+
   while true do
-    if instance.terminating_wait?
-      instance.continue_to_termination(hook_name: "TerminationHook")
-      break
-    elsif work_index_job.definitely_out_of_work?
-      instance.terminate(should_decrement_desired_capacity: true, continue_hook_name: "TerminationHook")
-      break
-    else
-      stats = work_index_job.call # reads from queue, works the job, writes to done queue
-      Rails.logger.info { "work_index_jobs #call w/ stats #{stats.to_s}" }
+    begin
+      if instance.terminating_wait?
+        Rails.logger.info { "work_index_jobs: in Terminating:Wait, continuing to Termination"}
+        instance.continue_to_termination(hook_name: "TerminationHook")
+        break
+      elsif work_index_job.definitely_out_of_work?
+        Rails.logger.info { "work_index_jobs: out of work, terminating"}
+        instance.terminate(should_decrement_desired_capacity: true, continue_hook_name: "TerminationHook")
+        break
+      else
+        Rails.logger.info { "work_index_jobs: starting #call"}
+        stats = work_index_job.call # reads from queue, works the job, writes to done queue
+        Rails.logger.info { "work_index_jobs: #call finished #{stats.to_s}" }
+      end
+    rescue Aws::AutoScaling::Errors::ScalingActivityInProgress => ee
```

Comment:
why ee?  usually, the general pattern for exceptions is 'ex' or 'exception' or even 'e'.  why a new one ee? :)

Reply:
Two Es are better than one.  Duh!

Reply:
I agree `e` was the original common way, but one letter variables are the root of all evil, so at some point in my distant past I doubled it to `ee`.